zerver/lib/user_groups.py

```diff
@@ -325,7 +325,9 @@ def get_recursive_subgroups(user_group: UserGroup) -> QuerySet[UserGroup]:
     cte = With.recursive(
         lambda cte: UserGroup.objects.filter(id=user_group.id)
         .values(group_id=F("id"))
-        .union(cte.join(UserGroup, direct_supergroups=cte.col.group_id).values(group_id=F("id")))
+        .union(
+            cte.join(NamedUserGroup, direct_supergroups=cte.col.group_id).values(group_id=F("id"))
+        )
     )
     return cte.join(UserGroup, id=cte.col.group_id).with_cte(cte)
```

Comment:
Is it correct that some of these remain `UserGroup`? I'm not entirely sure how to read this block.

Reply:
OK -- might be worth posting in #backend the SQL queries this generates and their `EXPLAIN ANALYZE` just for visibility, but probably will be fine.